test/gtest/uct/ib/test_dc.cc

```diff
@@ -440,6 +440,32 @@ UCS_TEST_P(test_dc, rand_dci_pending_purge) {
     flush();
 }
 
+UCS_TEST_P(test_dc, pool_indices)
+{
+    constexpr uint8_t max_dci_pools = UCT_DC_MLX5_IFACE_MAX_DCI_POOLS;
+    entity *rand_e;
+    uint8_t num_dci_pools;
+    uct_dc_mlx5_ep_t *ep;
+    uint8_t expected_pool_index;
+    int i;
+
+    /* Set num paths to max */
+    modify_config("IB_NUM_PATHS", std::to_string(max_dci_pools).c_str());
+
+    rand_e        = create_rand_entity();
```

Comment:
minor: could define vars like "auto num_dci_pools = dc_iface..." or "for (int i = 0; i < max_dci_pools..."